framework/src/timesteppers/IterationAdaptiveDT.C

```diff
@@ -258,14 +277,51 @@ IterationAdaptiveDT::computeFailedDT()
   return _dt * _cutback_factor;
 }
 
+bool
+IterationAdaptiveDT::converged()
+{
+  if (!_reject_large_step)
+    return TimeStepper::converged();
+
+  // the solver has not converged
+  if (!TimeStepper::converged())
+    return false;
+
+  // we are already at dt_min or at the start of the simulation
+  // in which case we can move on to the next step
+  if (_dt == _dt_min || _t_step < 2)
+    return true;
+
+  // we need to update the post-processor value
+  // (otherwise, _pps_value is the value computed at the previous time step)
+  if (_pps_value)
+    _fe_problem.execute(EXEC_TIMESTEP_END);
+
+  // we get what the next time step should be
+  Real dt_test = _dt;
+  limitDT(dt_test);
```

Comment:
Sorry, I'm a little late to the game in commenting on this. In general, I think this PR is a good idea. However, one thing about this is that it could reject the step based on things like iteration count (which is really driving the time step based on efficiency considerations), which I don't think we really want to do. It seems like the step should only be rejected if the postprocessor or function limits (which are based on accuracy considerations) are indicating a smaller time step. Maybe we could tweak this a little bit more to split out the functionality in `limitDT()` into two separate functions - one for accuracy and one for efficiency considerations, and then just call the accuracy one here.

Reply:
That is probably a good idea to avoid side effects (again, while the implementation works for the PP, I have not really tested it with the other methods). I am not sure how you want to proceed to fix this. Should we open a new issue or is there a way to continue work here?

Reply:
I'd just re-open the issue you were using.